UITests/Receipts/ListReceiptTests.swift

```diff
@@ -191,7 +194,166 @@ class ListReceiptTests: BaseTests {
     }
 
     private func openReceiptsListScreen() {
-        app.tables.cells.containing(.staticText, identifier: "List Receipts").element(boundBy: 0).tap()
+        app.tables.cells.containing(.staticText, identifier: "List User Receipts").element(boundBy: 0).tap()
         waitForNonExistence(spinner)
     }
+
+    // MARK: Detail Page Testcases
+
+    // Credit Transaction
+    func testReceiptDetail_verifyCreditTransaction() {
+        let expectedDateValue = "Fri, May 24, 2019, 6:16 PM"
+        openupReceiptsListScreenForFewMonths()
+        transactDetails.openReceipt(row: 0)
+        let transactionDetailHeaderLabel = transactDetails.detailHeaderTitle
+        waitForNonExistence(transactionDetailHeaderLabel)
+        if #available(iOS 12, *) {
+            verifyPayment(payment: "Payment\nMay 24, 2019", amount: "+6.00\n\(currency)")
+        } else {
+            verifyPayment(payment: "Payment May 24, 2019", amount: "+6.00 \(currency)")
+        }
+
+        // DETAILS Section
+        verifyDetailSection(receiptIdVal: "55176992", dateVal: expectedDateValue, clientIdVal: "DyClk0VG9a")
+
+        // FEE Section
+        verifyFeeSection(amountVal: "+6.00 USD", feeVal: "0.00 USD", transactionVal: "6.00 USD")
+    }
+
+    // Debit Transaction
+    func testReceiptDetail_verifyDebitTransaction() {
+        let expectedDateValue = "Sun, May 12, 2019, 6:16 PM" // Sun, May 12, 2019, 6:16 PM
+        openupReceiptsListScreenForFewMonths()
+        transactDetails.openReceipt(row: 1)
+        let transactionDetailHeaderLabel = transactDetails.detailHeaderTitle
+        waitForNonExistence(transactionDetailHeaderLabel)
+
+        if #available(iOS 12, *) {
+            verifyPayment(payment: "Bank Account\nMay 12, 2019", amount: "-5.00\n\(currency)")
+        } else {
+            verifyPayment(payment: "Bank Account May 12, 2019", amount: "-5.00 \(currency)")
+        }
+
+        // DETAILS Section
+        verifyDetailSection(receiptIdVal: "55176991", dateVal: expectedDateValue, clientIdVal: nil)
+
+        // FEE Section
+        verifyFeeSection(amountVal: "-5.00 USD", feeVal: "2.00 USD", transactionVal: "-7.00 USD")
+    }
+
+    func testReceiptDetail_verifyTransactionOptionalFields() {
+        openupReceiptsListScreenForOneMonth()
+        transactDetails.openReceipt(row: 4)
+        let transactionDetailHeaderLabel = transactDetails.detailHeaderTitle
+        waitForNonExistence(transactionDetailHeaderLabel)
```

Comment:
You want waitForExistence instead.  Details Header Title will be there after clicking the receipt row.